### PR TITLE
Add further debug logging for ds listing

### PR DESCRIPTION
**Pull request by MichaelBuessemeyer**
The method `listGrouped` needs to be investigated further.

> Answering 400 at /api/datasets?isActive=false for user \<id\>: {"messages":[{"error":"Grouping datasets failed"},{"chain":"[Server Time 2024-11-28 13:54]  <~ Failed to retrieve list of data sets."}]}

Follow up to #8246

### Steps to test:
- same as in #8246

### CODE REVIEW

File: app/controllers/DatasetController.scala [hunk 1]
```diff
@@ -224,28 +224,42 @@ class DatasetController @Inject()(userService: UserService,
       implicit ctx: DBAccessContext,
       m: MessagesProvider): Fox[List[JsObject]] =
     for {
+      _ <- Fox.successful(())
+      _ = logger.info(s"datasets: $datasets, requestingUser: $requestingUser")
```
Comment (fm3) [inline app/controllers/DatasetController.scala:228]:
```suggestion
      _ = logger.info(s"datasets: $datasets, requestingUser: ${requestingUser._id}")
```

**Review by fm3 — APPROVED** [review]
let’s avoid printing entire user objects, other than that LGTM

File: app/controllers/DatasetController.scala (hunk 1 above)
Comment (MichaelBuessemeyer) [reply app/controllers/DatasetController.scala]: > let’s avoid printing entire user objects,

only user objects?

Comment (fm3) [reply app/controllers/DatasetController.scala]: Yes, should be ok. This is only temporary, this isn’t super sensitive data and the logs are of course not public.

**Pull request merged**